backend/src/Routes/elections.routes.js

```diff
@@ -6,65 +6,42 @@ const electionController = require('../Controllers/elections.controllers')
     electionController.returnElection)
 router.post('/Election/:id/register',
-        authController.getUser,
-        voterRollController.getVoterAuth,
-        voterRollController.registerVoter)
+    authController.getUser,
+    voterRollController.getVoterAuth,
+    voterRollController.registerVoter)
 router.get('/Election/:id/ballots',
     authController.getUser,
     authController.hasPermission(permissions.canViewBallots),
     ballotController.getBallotsByElectionID,
     ballotController.returnBallots)
-router.get('/Election/:id/rolls',
-    authController.getUser,
-    authController.hasPermission(permissions.canViewElectionRoll),
-    voterRollController.getRollsByElectionID,
-    voterRollController.returnRolls)
-router.get('/Election/:id/rolls/:voter_id',
-    authController.getUser,
-    authController.hasPermission(permissions.canViewElectionRoll),
-    voterRollController.getByVoterID)
-router.post('/Election/:id/rolls/',
-    authController.getUser,
-    authController.hasPermission(permissions.canEditElectionRoll),
-    voterRollController.editElectionRoll)
-
-router.post('/Election/:id/rolls/approve',
-    authController.getUser,
-    authController.hasPermission(permissions.canApproveElectionRoll),
-    voterRollController.changeElectionRollState(ElectionRollState.approved))
-router.post('/Election/:id/rolls/flag',
-    authController.getUser,
-    authController.hasPermission(permissions.canFlagElectionRoll),
-    voterRollController.changeElectionRollState(ElectionRollState.flagged))
-router.post('/Election/:id/rolls/invalidate',
-    authController.getUser,
-    authController.hasPermission(permissions.canInvalidateElectionRoll),
-    voterRollController.changeElectionRollState(ElectionRollState.invalid))
-router.post('/Election/:id/rolls/unflag',
-    authController.getUser,
-    authController.hasPermission(permissions.canUnflagElectionRoll),
-    voterRollController.changeElectionRollState(ElectionRollState.invalid))
-router.post('/Election/:id/rolls/',
-    authController.getUser,
-    authController.hasPermission(permissions.canAddToElectionRoll),
-    voterRollController.addElectionRoll)
+router.get('/Election/:id/rolls', authController.getUser, getRollsByElectionID)
+router.get('/Election/:id/rolls/:voter_id', authController.getUser, getByVoterID)
+router.post('/Election/:id/rolls/', authController.getUser, editElectionRoll)
+router.post('/Election/:id/rolls/approve', authController.getUser, changeElectionRollStateController.approveElectionRoll)
+router.post('/Election/:id/rolls/flag', authController.getUser, changeElectionRollStateController.flagElectionRoll)
+router.post('/Election/:id/rolls/invalidate', authController.getUser, changeElectionRollStateController.invalidateElectionRoll)
+router.post('/Election/:id/rolls/unflag', authController.getUser, changeElectionRollStateController.uninvalidateElectionRoll)
+router.post('/Election/:id/rolls/', authController.getUser, addElectionRoll)
```

Comment:
In general, I'd like to see the controllers handle everything they need explicitly, rather than implicitly assuming that other methods have been chained in the route that stash data into the req.

Reply:
If it's really every route, then yes exactly we can use a middleware  (like you did for the catchError), and then add the user data explicitly to the IRequest (even if it's nullable).

If it's going to be conditional per route, I'd suggest putting it in a single function to handle the route, so that all the code required for the route is in one place.